src/matrices/skeletonization.jl

```diff
@@ -98,17 +98,20 @@ end
 # (close returns map) [obtained by calling first `revert_recurrence_matrix` on a
 # recurrence Matrix R, and consecutevely `horizontalhisto(R)`] construct and return
 # a "dummy" close returns map, where all lines are encoded according to their length.
-function create_close_returns_map(lines1::Vector{Int}, lines2::Vector{Int}, lines3::Vector{Int}, size_of_cl_ret_RP::Tuple{Int, Int})
-    X = zeros(Int, size_of_cl_ret_RP)
-    for i = 1:length(lines1)
-        line_length = lines1[i]
-        line_ind = lines2[i]
-        column_ind = lines3[i]
-        for j = 0:line_length-1
-            X[line_ind,column_ind+j] = line_length
-        end
+function create_close_returns_map(lines1, lines2, lines3, dims)
```

Comment:
The code of this function was reasonably readable and compact, but it created a full matrix to eventually convert it to a sparse one. I have modified it to make directly a sparse matrix, which should perform better when the matrix has many zeros.

I have also removed the type annotations of `create_close_returns_map` - which are not really necessary right now.